### PR TITLE
V1.6.0 with patches

### PR DESCRIPTION
**Pull request by leehinman**
Bring over the 2 commits that were in adriansr/fsnotify and resolve conflicts to use on version 1.6.0

Relates elastic/beats#40921

### CODE REVIEW

File: backend_fen.go [hunk 1]
```diff
@@ -160,3 +160,9 @@ func (w *Watcher) Add(name string) error {
 func (w *Watcher) Remove(name string) error {
 	return nil
 }
+
+// SetRecursive enables watches to also monitor subdirectories. Currently
+// only supported under Windows.
+func (w *Watcher) SetRecursive() error {
+	return nil
```
Comment (belimawr) [inline backend_fen.go:167]: [Question]
If it's only supported on Windows, why this one does not retunr an error?

File: backend_windows.go [hunk 2]
```diff
@@ -638,16 +648,8 @@ func (w *Watcher) readEvents() {
...
-			size := int(raw.FileNameLength / 2)
-			var buf []uint16
-			// TODO: Use unsafe.Slice in Go 1.17; https://stackoverflow.com/questions/51187973
-			sh := (*reflect.SliceHeader)(unsafe.Pointer(&buf))
-			sh.Data = uintptr(unsafe.Pointer(&raw.FileName))
-			sh.Len = size
-			sh.Cap = size
-			name := windows.UTF16ToString(buf)
+			buf := unsafe.Slice(&raw.FileName, raw.FileNameLength/2)
+			name := syscall.UTF16ToString(buf)
```
Comment (belimawr) [inline backend_windows.go:652]: [Question]
Why the change from `windows` to `syscall`?

File: backend_fen.go (hunk 1 above)
Comment (leehinman) [reply backend_fen.go:167]: Not the most satisfying answer, but because that is what the original code does.  It was in `fen.go`

https://github.com/adriansr/fsnotify/blob/a81f2b630e7c708d59a259e6308cd70409a1ebd3/fen.go#L39-L43

File: backend_windows.go (hunk 2 above)
Comment (leehinman) [reply backend_windows.go:652]: Again, not very satisfying.  But this what the original code does.

https://github.com/adriansr/fsnotify/blob/a81f2b630e7c708d59a259e6308cd70409a1ebd3/windows.go#L464-L465

Comment (leehinman) [reply backend_windows.go:652]: I'm ok with changing this later, but I think we should start with a copy of the code we are currently shipping.

Comment (belimawr) [reply backend_windows.go:652]: That makes sense.

**Review by belimawr — APPROVED** [review]
